src/app/components/ConnectorDrawer/ConnectorDrawer.tsx

```diff
@@ -82,87 +103,185 @@ export type ConnectorDrawerPanelContentProps = {
     error,
     onClose,
+    onConnectorDetail,
+    onDuplicateConnector,
   }) => {
+    const [showDeleteConnectorConfirm, setShowDeleteConnectorConfirm] =
+      useState(false);
     const { t } = useTranslation();
     const [activeTabKey, setActiveTabKey] = useState<string | number>(0);
 
     const selectActiveKey = (_: MouseEvent, eventKey: string | number) => {
       setActiveTabKey(eventKey);
     };
 
-    return (
-      <DrawerPanelContent widths={{ default: 'width_50' }}>
-        <DrawerHead>
-          <TextContent>
-            <Text
-              component={TextVariants.small}
-              className="connector-drawer__header-text"
-            >
-              {t('connectorName')}
-            </Text>
+    const [isOpen, setIsOpen] = useState<boolean>(false);
 
-            <Flex>
-              <FlexItem>
-                <Title
-                  headingLevel={'h2'}
-                  size={TitleSizes['xl']}
-                  className="connector-drawer__header-title"
-                >
-                  {name}
-                </Title>
-              </FlexItem>
-              <FlexItem spacer={{ default: 'spacerSm' }}>
-                <ConnectorStatus name={name} status={status} />
-              </FlexItem>
-            </Flex>
-          </TextContent>
-          <DrawerActions>
-            <DrawerCloseButton onClick={onClose} />
-          </DrawerActions>
-        </DrawerHead>
-        <DrawerPanelBody>
-          <Tabs activeKey={activeTabKey} onSelect={selectActiveKey}>
-            <Tab
-              eventKey={0}
-              title={<TabTitleText>{t('details')}</TabTitleText>}
-            >
-              <div className="connector-drawer__tab-content">
-                <ConnectorInfoTextList
-                  name={name}
-                  id={id}
-                  bootstrapServer={bootstrapServer}
-                  kafkaId={kafkaId}
-                  namespaceId={namespaceId}
-                  owner={owner}
-                  createdAt={createdAt}
-                  modifiedAt={modifiedAt}
-                  error={error}
-                />
-              </div>
-            </Tab>
-            {/* <Tab
+    const {
+      connector,
+      canStart,
+      canStop,
+      canDelete,
+      onStart,
+      onStop,
+      onDelete,
+    } = useConnector(currentConnectorRef as ConnectorMachineActorRef);
+    const onToggle = (isOpen: boolean) => {
+      setIsOpen(isOpen);
+    };
+    const onSelect = (
+      _event?: SyntheticEvent<HTMLDivElement, Event> | undefined
+    ) => {
+      setIsOpen(!isOpen);
+      onFocus();
+    };
+    const onFocus = () => {
+      const element = document.getElementById('connector-action');
+      element?.focus();
+    };
+    const doCancelDeleteConnector = () => {
+      setShowDeleteConnectorConfirm(false);
+    };
+    const doDeleteConnector = () => {
+      setShowDeleteConnectorConfirm(false);
+      onDelete();
+    };
+
+    const dropdownItems = [
```

Comment:
It might be nice to see if this list of actions could be defined in one place and then shared here.  It looks like in this case we don't need the "Details" link, but maybe a function or shared component that returns these?  This helps keep things consistent across the UI.

Reply:
@gashcrumb You are right, It would be nice to have a shared component. We might need this in the edit connectors page too. Thanks!